### PR TITLE
Simplify chain reorg detection code

### LINKED ISSUE #25: [BCL] check for chain reorgs with block hashes
When just comparing block numbers chain reorgs might be missed. Check for block hashes as well.

### PR DESCRIPTION
**Pull request by palango**
@hackaugusto The code already tested for reorgs with block hashes as well, but it wasn't obvious. I think I managed to simplify the logic a bit.

Closes #25

### CODE REVIEW

**Comment by pcppcp** [comment]
Do you also have a test for a reorg with different hash?
@hackaugusto regarding our last design discussion, are you ok with using this as transport interface with a wrapper around that implements app-specific functions?
`core <--> BlockchainImplementation <--> BlockchainTransport`

core can i.e. pull events from a queue fed from the transport...

**Comment by palango** [comment]
No, I don't see an easy way to create a reorg. The nearest would be to create a snapshot and revert to that to test at least one case.

**Comment by palango** [comment]
Nice, will add these next week.

**Comment by hackaugusto** [comment]
@pcppcp 

> regarding our last design discussion, are you ok with using this as transport interface with a wrapper around that implements app-specific functions?

I didn't really get the question

File: raiden_libs/blockchain.py [hunk 1]
```diff
@@ -242,41 +250,48 @@ def filter_events(self, filter_params: dict, name_to_callback: dict) -> None:
...
+                self._detected_chain_reorg(current_block)
+
+            # now we have to check that the confirmed_head_hash stayed the same
+            # otherwise the program aborts
             try:
-                self.web3.eth.getBlock(self.confirmed_head_hash)
-            except ValueError:
+                current_head_hash = self.web3.eth.getBlock(self.confirmed_head_number).hash
+                if current_head_hash != self.confirmed_head_hash:
+                    log.critical('Events considered confirmed have been reorganized')
```
Comment (pcppcp) [inline raiden_libs/blockchain.py]: Could you add more info to the log, block numbers and hashes in particular? It'll make it easier to find out what went wrong if this scenario happens.

**Pull request merged**